runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/FileWatcherTask.java

```diff
@@ -0,0 +1,132 @@
+package io.aklivity.zilla.runtime.engine.internal.registry;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_DELETE;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static org.agrona.LangUtil.rethrowUnchecked;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+
+public class FileWatcherTask extends WatcherTask
+{
+    private byte[] configHash;
+
+    public FileWatcherTask(URL configURL, Consumer<String> configChangeListener)
+    {
+        super(configURL, configChangeListener);
+    }
+
+    @Override
+    public boolean run()
+    {
+        try
+        {
+            doInitialConfiguration();
+
+            Path configPath = Paths.get(new File(configURL.getPath()).getAbsolutePath());
+
+            WatchService watchService = FileSystems.getDefault().newWatchService();
+            configPath.getParent().register(watchService, ENTRY_MODIFY, ENTRY_CREATE, ENTRY_DELETE);
```

Comment:
This call to `register` returns a `WatchKey`.

Suggest retaining a mapping of watched configURL paths to (parent) `WatchKey`. This will help with the test below when a `WatchKey` is returned from `watchService.take()`. If the `event.context()` path is not present then the event can be ignored.

We'd also need to retain a mapping of watched configURL paths to computed hash value.

The change listener callback signature will need to change to `(URL configURL, String oldValue, String newValue)`.